controllers/managedocs_controller_test.go

```diff
@@ -434,11 +434,96 @@ var _ = Describe("ManagedOCS controller", func() {
 				Expect(configMap.Data["test-key"]).Should(Equal("test-value"))
 			})
 		})
+		When("there is no enable-mcg field in the add-on parameters secret", func() {
+			It("should not create reconciled resources", func() {
+				Skip("Skipping this test till Nooba MCG is integrated in deployer")
+				secret := addonParamsSecretTemplate.DeepCopy()
+				secret.Data["size"] = []byte("1")
+				Expect(k8sClient.Create(ctx, secret)).Should(Succeed())
+				// Ensure, over a period of time, that the resource is not created
+				utils.EnsureNoResource(k8sClient, ctx, scTemplate.DeepCopy(), timeout, interval)
+			})
+		})
+		When("there is an invalid enable-mcg value in the add-on parameters secret", func() {
+			It("should not create reconciled resources", func() {
+				// Create a invalid enable-mcg parameter value
+				secret := addonParamsSecretTemplate.DeepCopy()
+				secret.Data["size"] = []byte("1")
+				secret.Data["enable-mcg"] = []byte("NO")
+				Expect(k8sClient.Create(ctx, secret)).Should(Succeed())
+
+				// Ensure, over a period of time, that the resource is not created
+				utils.EnsureNoResource(k8sClient, ctx, scTemplate.DeepCopy(), timeout, interval)
+			})
+		})
+		When("there is a valid enable-mcg value in the addon-on parameters secret which is false", func() {
+			It("should not change the storage cluster spec", func() {
```

Comment:
Again this is not specific enough
```suggestion
			It("Should change storagecluster's MCG reconcile strategy to manage ", func() {
```